convert/gitlab/convert.go

```diff
@@ -256,6 +326,278 @@ func convertJobToStep(ctx *context, jobName string, job *gitlab.Job) []*harness.
 	return steps
 }
 
+// convertInheritDefaultFields converts the default fields from the default job into the current job.
+func convertInheritDefaultFields(spec *harness.StepExec, defaultJob *gitlab.Default, keys []string) {
+	for _, key := range keys {
+		switch key {
+		case "after_script":
+			if len(defaultJob.After) > 0 {
+				spec.Run = strings.Join(defaultJob.After, "\n")
+			}
+		case "before_script":
+			if len(defaultJob.Before) > 0 {
+				spec.Run = strings.Join(defaultJob.Before, "\n")
+			}
+		case "artifacts":
+			if defaultJob.Artifacts != nil {
+				//TODO no supported
+			}
+		case "cache":
+			if defaultJob.Cache != nil {
+				//TODO
+			}
+		case "image":
+			if defaultJob.Image != nil {
+				spec.Image, spec.Pull = convertImageAndPullPolicy(defaultJob.Image)
+			}
+		case "interruptible":
+			//TODO not supported
+		case "retry":
+			if defaultJob.Retry != nil {
+				//TODO not supported
+			}
+		case "services":
+			if len(defaultJob.Services) > 0 {
+				//TODO
+			}
+		case "tags":
+			if len(defaultJob.Tags) > 0 {
+				//spec.Tags = strings.Join(defaultJob.Tags, ", ") //TODO
+			}
+		case "duration":
+			//spec.Timeout = defaultJob.Timeout //TODO
+		default:
+			log.Printf("Unrecognized inherit key: %s", key)
+		}
+	}
+}
+
+// convertInherit converts the inherit fields from the default job into the current job.
+func convertInheritedVariables(job *gitlab.Job, stageEnvs map[string]string) map[string]string {
+	if job.Inherit == nil || job.Inherit.Variables == nil {
+		return stageEnvs
+	}
+
+	if job.Inherit.Variables.All {
+		return stageEnvs
+	}
+
+	// If inherit.variables is an array, only the variables in the array are inherited.
+	if job.Inherit.Variables.Keys != nil {
+		newEnvs := make(map[string]string)
+		for _, key := range job.Inherit.Variables.Keys {
+			if value, ok := stageEnvs[key]; ok {
+				newEnvs[key] = value
+			}
+		}
+		return newEnvs
+	}
+
+	return stageEnvs
+}
+
+// convertImageAndPullPolicy converts a GitLab image to a Harness image and pull policy.
+func convertImageAndPullPolicy(image *gitlab.Image) (string, string) {
+	name := ""
+	pullPolicy := ""
+
```

Comment:
the above syntax is correct, but from a style perspective you will typically see these declared as zero value variables:

```
var name string
var pullPolicy string
```

just something to consider in the future

Reply:
Yes that is cleaner, I pushed this change